`hickory-server` `+dnssec-ring` does not build with Rust 1.79

**Describe the bug**
What the title says. As I understand, the current MSRV is 1.70 as [per CI](https://github.com/hickory-dns/hickory-dns/blob/9ad0cf2590ee6fd5a2ac0237fa417b9b30555bea/.github/workflows/test.yml#L127).

**To Reproduce**
Steps to reproduce the behavior:

``` console
$ cargo +1.79.0 build -p hickory-server --features dnssec-ring
(..)
error[E0277]: the trait bound `[u8]: Default` is not satisfied
   --> crates/server/src/config/dnssec.rs:262:9
    |
262 |         /// The salt used for hashing.
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `Default` is not implemented for `[u8]`, which is required by `Arc<[u8]>: Default`
    |
    = help: the following other types implement trait `Default`:
              &[T]
              &mut [T]
              [T; 0]
              [T; 10]
              [T; 11]
              [T; 12]
              [T; 13]
              [T; 14]
            and 27 others
    = note: required for `Arc<[u8]>` to implement `Default`

```

**Expected behavior**
The code should compile with that toolchain version.

**System:**
 - OS: Ubuntu
 - Architecture: x86_64
 - Version 22.04
 - rustc version: 1.79.0

**Version:**
Crate: `hickory-server`
Version: 9ad0cf2590ee6fd5a2ac0237fa417b9b30555bea

--- a/crates/server/src/lib.rs
+++ b/crates/server/src/lib.rs
@@ -82,7 +82,7 @@ pub mod dnssec {
             #[serde(default)]
             algorithm: Nsec3HashAlgorithm,
             /// The salt used for hashing.
-            #[serde(default)]
+            #[serde(default = "default_salt")]
             salt: Arc<[u8]>,
             /// The number of hashing iterations.
             #[serde(default)]
@@ -91,6 +91,11 @@ pub mod dnssec {
             #[serde(default)]
             opt_out: bool,
         },
+    }
+
+    // MSRV: works in 1.80, fails in 1.78
+    fn default_salt() -> Arc<[u8]> {
+        Arc::new([])
     }
 }
 

